src/main/java/org/jenkinsci/plugins/workflow/multibranch/SCMVar.java

```diff
@@ -75,20 +78,30 @@
             throw new IllegalStateException("inappropriate context");
         }
         Branch branch = property.getBranch();
+        ItemGroup<?> parent = job.getParent();
+        if (!(parent instanceof WorkflowMultiBranchProject)) {
+            throw new IllegalStateException("inappropriate context");
+        }
+        SCMSource scmSource = ((WorkflowMultiBranchProject) parent).getSCMSource(branch.getSourceId());
+        if (scmSource == null) {
+            throw new IllegalStateException(branch.getSourceId() + " not found");
+        }
+        SCMRevision tip;
         SCMRevisionAction revisionAction = build.getAction(SCMRevisionAction.class);
         if (revisionAction != null) {
-            ItemGroup<?> parent = job.getParent();
-            if (!(parent instanceof WorkflowMultiBranchProject)) {
-                throw new IllegalStateException("inappropriate context");
-            }
-            SCMSource scmSource = ((WorkflowMultiBranchProject) parent).getSCMSource(branch.getSourceId());
-            if (scmSource == null) {
-                throw new IllegalStateException(branch.getSourceId() + " not found");
-            }
-            return scmSource.build(branch.getHead(), revisionAction.getRevision());
+            tip = revisionAction.getRevision();
         } else {
-            return branch.getScm();
+            SCMHead head = branch.getHead();
+            TaskListener listener = TaskListener.NULL; // TODO would like to send to build log, but have no way of doing that here
+            tip = scmSource.fetch(head, listener);
+            if (tip == null) {
+                // TODO would it be better to throw AbortException? Cf. SCMBinder.
```

Comment:
Yes; for purposes of this PR I wanted to avoid making changes that are not really related to the refactoring. You can see the existing behavior in the test for the case of a deleted branch. Probably a failure here (and in the analogous code in `SCMBinder`) would be clearer, assuming there are no _legitimate_ cases where `fetch` would return null here. But best done in a follow-up.